test/test.js

```diff
@@ -0,0 +1,141 @@
+const mocha = require('mocha')
+const { describe, it } = mocha
+
+const assert = require('assert')
+
+const windows = require('..')('1.8')
+const Item = require('prismarine-item')('1.8')
+
+const chest = windows.createWindow(1, 'minecraft:chest', 'chest', 27)
+const inv = windows.createWindow(2, 'minecraft:inventory', 'inventory')
+
+const createClick = (mode, mouseButton, slot, useInv = false) => ({
+  slot,
+  mouseButton,
+  mode,
+  windowId: useInv ? inv.id : chest.id,
+  item: slot === -999 ? null : useInv ? inv.slots[slot] : chest.slots[slot]
+})
+
+chest.updateSlot(0, new Item(1, 64))
+chest.updateSlot(1, new Item(1, 1))
+chest.updateSlot(3, new Item(1, 63))
+chest.updateSlot(5, new Item(2, 3))
+chest.updateSlot(chest.inventoryEnd - 1, new Item(1, 44))
+chest.updateSlot(chest.inventoryEnd - 2, new Item(1, 44))
+chest.updateSlot(chest.inventoryEnd - 3, new Item(1, 44))
+
+inv.updateSlot(inv.inventoryEnd - 1, new Item(301, 1))
+inv.updateSlot(inv.inventoryEnd - 2, new Item(1, 1))
```

Comment:
So the chest content and inventory content is initialized as the test is loaded. What happens when clicks are accepted? Do the items move inside of the container? Maybe it is better to use mocha's `beforeEach()` hook? https://mochajs.org/#hooks

You can run specific tests with mocha by using regex to match the test name. Do the tests break if they are not all executed after each other?

Reply:
yes, the tests absolutely suck and will break if they are not executed in order xD

i have no idea how mocha works, just wanted to make minimal tests, which show that my changes work.
if you understand how the logic behind the tests works, then you are likely able to create a better test.js